engine/src/main/java/org/terasology/rendering/dag/nodes/OutputToScreenNode.java

```diff
@@ -34,29 +36,85 @@
 
+        displayResolutionDependentFBOs = context.get(DisplayResolutionDependentFBOs.class);
+        worldRenderer = context.get(WorldRenderer.class);
+
         WorldRenderer worldRenderer = context.get(WorldRenderer.class);
         requiresCondition(() -> worldRenderer.getCurrentRenderStage() == MONO || worldRenderer.getCurrentRenderStage() == LEFT_EYE);
 
         addDesiredStateChange(new EnableMaterial(DEFAULT_TEXTURED_MATERIAL_URN));
 
-        addDesiredStateChange(new SetInputTextureFromFbo(0, FINAL_BUFFER, ColorTexture,
-                context.get(DisplayResolutionDependentFBOs.class), DEFAULT_TEXTURED_MATERIAL_URN, "texture"));
+        bindFbo = new SetInputTextureFromFbo(0, FINAL_BUFFER, ColorTexture, displayResolutionDependentFBOs, DEFAULT_TEXTURED_MATERIAL_URN, "texture");
+        addDesiredStateChange(bindFbo);
+
+        SwappableFBO gBufferPair = displayResolutionDependentFBOs.getGBufferPair();
+        lastUpdatedGBuffer = gBufferPair.getLastUpdatedFbo();
+        staleGBuffer = gBufferPair.getStaleFbo();
     }
 
     @Override
     public void process() {
-        PerformanceMonitor.startActivity("rendering/copyImageToScreen");
+        PerformanceMonitor.startActivity("rendering/outputToScreen");
         // The way things are set-up right now, we can have FBOs that are not the same size as the display (if scale != 100%).
         // However, when drawing the final image to the screen, we always want the viewport to match the size of display,
         // and not that of some FBO. Hence, we are manually setting the viewport via glViewport over here.
         glViewport(0, 0, Display.getWidth(), Display.getHeight());
         renderFullscreenQuad();
         PerformanceMonitor.endActivity();
     }
+
+    @Override
+    public void handleCommand(String command, String... arguments) {
+        switch (command) {
+            case "setFbo":
+                if (arguments.length != 1) {
+                    throw new RuntimeException("Invalid number of arguments; expected 1, received " + arguments.length + "!");
+                }
+
+                FBO fbo;
+                switch (arguments[0]) {
+                    case "engine:fbo.gBuffer":
+                    case "engine:fbo.lastUpdatedGBuffer":
+                        fbo = lastUpdatedGBuffer;
+                        break;
+                    case "engine:fbo.staleGBuffer":
+                        fbo = staleGBuffer;
+                        break;
+                    default:
+                        fbo = displayResolutionDependentFBOs.get(new SimpleUri(arguments[0]));
+
+                        if (fbo == null) {
+                            throw new RuntimeException("Invalid fbo uri: " + arguments[0] + "!");
+                        }
+
+                        break;
+                }
+                setFbo(fbo);
+
+                break;
+            default:
+                throw new RuntimeException("Invalid command: " + command + "!");
```

Comment:
I'm not sure what happened here. This line originally had the following comment associated with it:

`// TODO: Throw error or log something?`

To which I replied (8 days ago):
> It should simply print an error message back to the console. Failing that, a log warning will be enough for the time being. But the end game here will always be to inform the user back in the console given that's where the command started.

Why have you added exceptions then? Exceptions are for cases where a part of the code can't handle a problem and escalates it to the caller, potentially all the way to the user. 

But here the code can handle the situation reasonably well: send a message back to the console that the command cannot successfully execute (with reasons) and do nothing. If that's too difficult for this PR, just send a warning to the log that the command couldn't execute, again with reasons, and return from the method doing nothing.